main.go

```diff
@@ -134,23 +138,65 @@ func updateNDK(revision string) error {
 		return err
 	}
 
+	if err := updateNDKPathIfNeeded(ndkHome); err != nil {
+		return err
+	}
+
 	if !inPath(ndkHome) {
 		log.Printf("Append to $PATH")
 		if err := tools.ExportEnvironmentWithEnvman("PATH", fmt.Sprintf("%s:%s", os.Getenv("PATH"), ndkHome)); err != nil {
 			return err
 		}
 	}
 
-	if os.Getenv("ANDROID_NDK_HOME") == "" {
-		log.Printf("Export ANDROID_NDK_HOME: %s", ndkHome)
-		if err := tools.ExportEnvironmentWithEnvman("ANDROID_NDK_HOME", ndkHome); err != nil {
+	if os.Getenv(androidNDKHome) == "" {
+		log.Printf("Export %s: %s", androidNDKHome, ndkHome)
+		if err := tools.ExportEnvironmentWithEnvman(androidNDKHome, ndkHome); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
+func updateNDKPathIfNeeded(ndkHome string) error {
+	log.Printf("searching for platforms dir in %s", ndkHome)
+	var foundPDPath string
+	if err := filepath.Walk(ndkHome, func(currentPath string, info os.FileInfo, e error) error {
+		if e != nil {
+			return e
+		}
+
+		if info.IsDir() && info.Name() == platformDirName {
+			foundPDPath = path.Dir(currentPath)
```

Comment:
currentPath always will be the path relative to `filepath.Walk(ndkHome` <- ndkHome, please make sure you have these joined -> `filepath.Join(ndkHome, currentPath)`

Reply:
also ndkpath this way will never == with the foundPDPath

Reply:
As it turned out, when you give it an absolute path, it uses an absolute path, if a relative, it uses relative path.